src/firebase/test/gcp-util.test.js

```diff
@@ -0,0 +1,26 @@
+import { uploadSelfImage, observeStimuliCompletion } from '../api/gcp-utils';
+import { useState } from 'react';
+
+test('upload self image', () => {
+  const userId = 'test';
+  const experimentId = '001';
+  const image = new File('../../../../testImages/test01.jpeg');
+  expect(uploadSelfImage(userId, experimentId, image)).toEqual({
+    status: StatusCodes.CREATED,
+    message: 'image successfully uploaded',
+    data: null,
+  });
+});
+
+test('observe stimuli completion', async () => {
+  const userId = 'test';
+  const experimentId = '001';
+  const [urls, setUrls] = useState([]);
+  const [error, setError] = useState('');
+  const imageUrlsHandler = (urls) => setUrls(urls);
+  const errorHandler = (error) => setError(error);
+  await observeStimuliCompletion(userId, experimentId,
+    imageUrlsHandler, errorHandler);
+  expect(urls).toHaveLength(400);
+  expect(error).toBe('');
+});
```

Comment:
do these tests actually upload an image and generate stimuli?

Reply:
Not yet, while I am thinking since this unit test will be called every time in Travis CI for any PR, shall we limit this?
Also I noticed that our cloud storage has already exceeded the free 5GB limit.

Reply:
my point is this should not upload an image and generate stimuli since this will no longer be unit test?

> Also I noticed that our cloud storage has already exceeded the free 5GB limit.
ah nice catch. That's filled up so fast. Imma delete some stuff

Reply:
> my point is this should not upload an image and generate stimuli since this will no longer be unit test?
> 
> > Also I noticed that our cloud storage has already exceeded the free 5GB limit.
> 
> ah nice catch. That's filled up so fast. Imma delete some stuff

We found this in our meeting when we were talking about budget analysis. This seems like the only major cost besides the cost of hosting the site. It will skyrocket the cost if we save the generated stimuli in Firestore.

Reply:
Yes since every time we upload an image the stimuli is generated, 
I think I can just do a unit test for the function of getting image urls from cloud storage.